app/src/main/java/com/github/se/wanderpals/MainActivity.kt

```diff
@@ -118,83 +118,76 @@ class MainActivity : ComponentActivity() {
-              } else if (navigationActions.variables.currentAddress != "") {
-                Trip(
-                    navigationActions,
-                    navigationActions.variables.currentTrip,
-                    tripsRepository,
-                    placesClient,
-                    Route.MAP)
-              } else {
-                Trip(
-                    navigationActions,
-                    navigationActions.variables.currentTrip,
-                    tripsRepository,
-                    placesClient)
-              }
-            }
-            composable(Route.CREATE_TRIP) {
-              BackHandler(true) {}
-              CreateTrip(OverviewViewModel(tripsRepository), navigationActions)
-            }
+          globalVariables = NavigationActionsVariables()
+          navigationActions =
+              NavigationActions(
+                  mainNavigation = rememberMultiNavigationAppState(startDestination = ROOT_ROUTE),
+                  tripNavigation =
+                      rememberMultiNavigationAppState(startDestination = Route.DASHBOARD))
+
+          NavHost(
+              navController = navigationActions.mainNavigation.getNavController,
+              startDestination = Route.SIGN_IN,
+              route = ROOT_ROUTE) {
+                composable(Route.SIGN_IN) {
+                  SignIn(
+                      onClick1 = { launcher.launch(signInClient.signInIntent) },
+                      onClick2 = {
+                        tripsRepository = TripsRepository(it.hashCode().toString(), Dispatchers.IO)
+                        tripsRepository.initFirestore(FirebaseApp.initializeApp(context)!!)
+                        val displayName = it.substringBefore('@')
+                        SessionManager.setUserSession(
+                            userId = it.hashCode().toString(), name = displayName, email = it)
+                        navigationActions.navigateTo(Route.OVERVIEW)
+                      })
+                }
+                composable(Route.OVERVIEW) {
+                  Overview(
+                      overviewViewModel = OverviewViewModel(tripsRepository),
+                      navigationActions = navigationActions)
+                }
+                composable(Route.TRIP) {
+                  navigationActions.tripNavigation.setNavController(rememberNavController())
+                  val routeToGo =
+                      if (navigationActions.variables.suggestionId != "") {
+                        Route.SUGGESTION
+                      } else if (navigationActions.variables.currentAddress != "") {
+                        Route.MAP
+                      } else {
+                        Route.DASHBOARD
+                      }
+                  val tripId = navigationActions.variables.currentTrip
+                  navigationActions.tripNavigation.setNavController(rememberNavController())
+                  navigationActions.tripNavigation.setStartDestination(routeToGo)
```

Comment:
Instead of modifying the navactions start destination, you can directly create a variable in the navactions that remembers which navcontroller was used so that, when you go back, you know which graph was used.